### DISCUSSION TITLE
Can't see Bubblejail after running a game in Lutris

### DISCUSSION THREAD

**Discussion by HowToCount** [igo95862/bubblejail#122] [Q&A]
I created an instance for Lutris, ran a game, and didn't see Bubblejail/Bubblewrap in the game process. When I tried using Firejail by setting the command prefix in Lutris, everything worked fine. So I assume Bubblejail only affects Lutris itself, but when I launch a game from Lutris, it starts as usual. How can I fix this?
I used "ps aux | grep" to check processes.

**Comment by igo95862** [comment]
Hello @ProgKing 

I am not sure what you mean with "didn't see Bubblejail/Bubblewrap in the game process." Could you please elaborate?

**Reply by igo95862** [reply]
If you want to check if the game is running inside sandbox you can for example use `lsns`. `lsns --tree  --task <game_pid>` will give you the tree of namespaces the game process is nested in.

You can also run a terminal inside the sandbox: `bubblejail run <lutris_instance_name> <alaritty_or_your_choice_of_terminal>` In the terminal you can run `ps aux` and your game should be one of the few running processes.

**Comment by HowToCount** [comment]
Hi @igo95862,

Sorry for the delay in getting back to you. I’ve tried various solutions but am still encountering errors. I’m using Arch Linux with Hyprland, and I also tested on Qtile X11.

When I try to execute `bubblejail run net.lutris.Lutris`
I receive this output:
`Instance already running.
Sending command to the instance:  []
Traceback (most recent call last):
  File "/usr/bin/bubblejail", line 20, in <module>
    bubblejail_main()
  File "/usr/lib/bubblejail/python-packages/bubblejail/bubblejail_cli.py", line 232, in bubblejail_main
    func(**args_dict)
  File "/usr/lib/bubblejail/python-packages/bubblejail/bubblejail_cli.py", line 83, in run_bjail
    command_return_text = async_run(
                          ^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/lib/bubblejail/python-packages/bubblejail/bubblejail_instance.py", line 152, in send_run_rpc
    (reader, writer) = await open_unix_connection(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/streams.py", line 97, in open_unix_connection
    transport, _ = await loop.create_unix_connection(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/unix_events.py", line 261, in create_unix_connection
    await self.sock_connect(sock, path)
  File "/usr/lib/python3.12/asyncio/selector_events.py", line 651, in sock_connect
    return await fut
           ^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/selector_events.py", line 659, in _sock_connect
    sock.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused`

Attempting to run Lutris (net.lutris.Lutris bubble) results in the following error:
`Failed to run instance: net.lutris.Lutris Exception: ConnectionRefusedError: [Errno 111] Connection refused`

I managed to open Lutris via Bubblejail once, but I haven't been able to open it again

**Reply by igo95862** [reply]
Looks like bubblejail instance crashed and did not clean-up properly. Then a new command is invoked it tries to send the argument to a running instance but it cannot connect to the socket.

You can force clean-up by deleting `/run/user/<user_id>/bubblejail/<instance_name>` directory but there probably will be dangling subprocesses in the system until it reboots.

**Reply by igo95862** [reply]
Could you investigate why it crashes and does not clean-up properly?

**Comment by HowToCount** [comment]
For some reason, it was fixed after rebooting. But now I have new issues, I opened Lutris and tried to launch a few games, and encountered these errors:

```
GameMode ERROR: D-Bus error: Could not call method 'RegisterGame' on 'com.feralinteractive.GameMode': org.freedesktop.DBus.Error.ServiceUnknown
gamemodeauto: D-Bus error: Could not call method 'RegisterGame' on 'com.feralinteractive.GameMode': org.freedesktop.DBus.Error.ServiceUnknown
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  150 (GLX)
  Minor opcode of failed request:  3 (X_GLXCreateContext)
  Value in failed request:  0x0
  Serial number of failed request:  277
  Current serial number in output stream:  278
GameMode ERROR: D-Bus error: Could not call method 'UnregisterGame' on 'com.feralinteractive.GameMode': org.freedesktop.DBus.Error.ServiceUnknown
gamemodeauto: D-Bus error: Could not call method 'UnregisterGame' on 'com.feralinteractive.GameMode': org.freedesktop.DBus.Error.ServiceUnknown
Monitored process exited.
```

And with another game this issue:
```
Setting scheduler resolution to 1
[STEAMWORKS]: SteamAPI_Init had succeeded without errors, debug flag = 0
YYExtensionInitialise CONFIGURED 
 Ok, you can continue
SteamUser()->BLoggedOn()[RunnerExperiments.dll] Successfully initialized runner interface
Could not find function "ImSetCurrentContext" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "SetClipboardText" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "IOSetDisplayFramebufferScaling" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "IOGetDisplayFramebufferScaling" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "IOSetINIFilename" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "IOGetINIFilename" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "PushClipRectFullscreen" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Could not find function "TextEditorsCleanup" in "C:\Games\Norland v0.3984.5638.1a\Imguigml_x64.dll" with argument types [].
Attempting to set gamepad count to 12
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  150 (GLX)
  Minor opcode of failed request:  3 (X_GLXCreateContext)
  Value in failed request:  0x0
  Serial number of failed request:  340
  Current serial number in output stream:  341
Monitored process exited.
Initial process has exited (return code: 256)
Exit with return code 256
```

I have no idea why GameMode has problems with D-Bus. Also, in the instance settings, I added a path to a directory with games. I'm not sure if this affects anything in my case, but I thought I'd mention it. Sorry, I have no idea what to do; I've never tried using sandboxes before.

**Reply by igo95862** [reply]
By the way, can you post logs with the triple ticks `\```\` instead of single tick so that the newlines are preserved and are easier to read?

**Reply by HowToCount** [reply]
> By the way, can you post logs with the triple ticks \```\ instead of single tick

Sure, sorry ;) 
I’m also a newbie on GitHub

I decided to read the journal, and I can see repetitive errors like this:
```Aug 26 22:38:42 archBTW gamemoded[114348]: ERROR: Failed to open file for read /sys/class...
Aug 26 22:38:42 archBTW gamemoded[114348]: ERROR: Could not inspect tasks for client [137...
Aug 26 22:38:42 archBTW gamemoded[114348]: ERROR: Failed to open file for read /sys/class...
Aug 26 22:38:42 archBTW gamemoded[114348]: ERROR: Failed to open file for read /sys/class/powercap/intel-rapl/intel-rapl:0/intel-rapl:powercap/intel-rapl/intel-rapl:0/intel-rapl:0:0/energy_uj
Aug 26 22:38:42 archBTW gamemoded[114348]: ERROR: Failed to open file for read /sys/class...
Aug 26 22:38:45 archBTW gamemoded[114348]: ERROR: Could not inspect tasks for client [137...
Aug 26 22:38:47 archBTW gamemoded[114348]: ERROR: Failed to open file for read /sys/class...
Aug 26 22:38:48 archBTW gamemoded[114348]: ERROR: Failed to open file for read /sys/class...```

**Reply by igo95862** [reply]
I will look in to gamemode. It seems to use some kind of D-Bus API.

As for `X Error of failed request:  BadValue (integer parameter out of range for operation)` it seems to be a driver issue? https://stackoverflow.com/questions/40644455/x-error-of-failed-request-badvalue-integer-parameter-out-of-range-for-operatio
